### PR TITLE
feat(integrations): Added a client call to get a single commit

### PR DESCRIPTION
**Pull request by AniketDas-Tekky**
Gitlab client can now retrieve a single commit.

### CODE REVIEW

File: src/sentry_plugins/github/client.py [hunk 1]
```diff
@@ -29,6 +29,11 @@ def get_pr_commits(self, repo, num):
         # Max: 250 Commits
         return self.get(f"/repos/{repo}/pulls/{num}/commits")
 
+    def get_commit_details(self, repo, sha):
```
[the remark this answers is not in the document]
Comment (AniketDas-Tekky) [reply src/sentry_plugins/github/client.py:32]: argh
MB

**Review by ceorourke — COMMENTED** [review]
Can you add a test for this?

**Comment by ceorourke** [comment]
Also is this just a test and it'll be added to GH and GL later on?

**Comment by AniketDas-Tekky** [comment]
GH client already has a get_commit API. Adding one for GL now

File: src/sentry/integrations/bitbucket/client.py [hunk 2]
```diff
@@ -120,6 +120,11 @@ def get_last_commits(self, repo, end_sha):
         data = self.get(BitbucketAPIPath.repository_commits.format(repo=repo, revision=end_sha))
         return self.zip_commit_data(repo, data["values"])
 
+    def get_commit(self, repo, sha):
+        # return api request that fetches the commit details
+        # see https://developer.atlassian.com/cloud/bitbucket/rest/api-group-commits/#api-repositories-workspace-repo-slug-commit-commit-get
+        return self.get(f"/2.0/repositories/{repo}/commits/{sha}")
```
Comment (scefali) [inline src/sentry/integrations/bitbucket/client.py:126]: @AniketDas-Tekky why not utilize a cache here?

File: src/sentry/integrations/gitlab/client.py [hunk 3]
```diff
@@ -258,6 +258,13 @@ def get_last_commits(self, project_id, end_sha):
         path = GitLabApiClientPath.commits.format(project=project_id)
         return self.get(path, params={"until": end_date})
 
+    def get_commit(self, project_id, sha):
+        """
+        Get the details of a commit
+        See https://docs.gitlab.com/ee/api/commits.html#get-a-single-commit
+        """
+        return self.get(GitLabApiClientPath.commit.format(project=project_id, sha=sha))
```
Comment (scefali) [inline src/sentry/integrations/gitlab/client.py:266]: @AniketDas-Tekky can we use get_cached here?

Comment (AniketDas-Tekky) [reply src/sentry/integrations/gitlab/client.py:266]: mb, I updated the wrong client

Comment (scefali) [reply src/sentry/integrations/gitlab/client.py:266]: @AniketDas-Tekky but we can still use `get_cached`

Comment (AniketDas-Tekky) [reply src/sentry/integrations/gitlab/client.py]: updated

File: src/sentry/integrations/bitbucket/client.py [hunk 4]
```diff
@@ -26,6 +26,7 @@ class BitbucketAPIPath:
 
     repository = "/2.0/repositories/{repo}"
     repositories = "/2.0/repositories/{username}"
+    repository_commit = "/2.0/repositories/{repo}/commit/{sha}"
```
Comment (ceorourke) [inline src/sentry/integrations/bitbucket/client.py:29]: Are the BitBucket endpoints used anywhere?

Comment (AniketDas-Tekky) [reply src/sentry/integrations/bitbucket/client.py:29]: Not at the moment, but I plan to followup with one

**Review by ceorourke — APPROVED** [review]

**Pull request merged**